js/script.js

```diff
@@ -0,0 +1,13 @@
+
+var allButtons = document.getElementsByTagName('button');
+
+console.log(allButtons);
+
+var targetButton = document.getElementsByClassName('button');
+	
+console.log(targetButton);
+console.log(targetButton.length);
+for ( var i = 0 ; i < (targetButton.length); i++) {
```

Comment:
move `targetButton.length` to new variable and use it in loop. in this case you calc length of array every time when conditions are true